js/components/common/react_modal.js.jsx

```diff
@@ -23,41 +23,36 @@ class CourseModal extends React.Component {
       courseId: '',
       course: [],
       sessions: [],
-      modalIsOpen: false,
       courseTitle: '',
     };
-    this.openModal = this.openModal.bind(this);
     this.closeModal = this.closeModal.bind(this);
   }
 
-  openModal(newCourse) {
-    if (newCourse === this.state.courseId) {
-      this.setState({ modalIsOpen: true });
-    } else {
-      getCourse(newCourse)
+  closeModal() {
+    this.props.onClose();
+  }
+
+  componentDidUpdate (prevProps) {
+    if ((prevProps.courseId !== this.props.courseId) && (this.props.courseId !== "")) {
+      getCourse(this.props.courseId)
         .then(course => {
-            //Tutorials don't have a timeStr to print, so I've currently omitted them
+          // Tutorials don't have a timeStr to print, so I've currently omitted them
           this.setState({
             course: course,
             sessions: course.allMeetingTimes.sort((firstLec, secondLec) =>
               firstLec.meetData.session > secondLec.meetData.session ? 1 : -1),
-            modalIsOpen: true,
-            courseId: newCourse,
-            courseTitle: `${newCourse.toUpperCase()} ${course.title}`
+            courseId: this.props.courseId,
```

Comment:
You can remove the `courseId` state entirely (in general, do not use state that duplicates a value from a prop).

Reply:
@EclipseIsDead I think you missed this comment.

Reply:
That's odd I swear I saw it in git diff. I implemented it this time though.